app/src/main/java/com/dimagi/biometric/activities/VerifyActivity.java

```diff
@@ -33,4 +33,16 @@ protected void onCaptureCancelled() {
                 caseId, new MatchResult(0, MatchStrength.ONE_STAR)
         ).finalizeResponse(this);
     }
+
+    @Override
+    protected ArrayList<String> validateRequiredParams() {
+        ArrayList<String> errors = new ArrayList<>();
+        if (caseId == null) {
+            errors.add(getText(R.string.missing_case_id).toString());
+        }
+        if (templateStr == null) {
```

Comment:
I don't see where `templateStr` is declared. Did I miss it?

Reply:
Ah, I seem to have by mistake put the declaration of this variable in a [later PR](https://github.com/dimagi/biometric-android/pull/6). These should all be merged together, so this should be fine.